infrastructure/src/main/java/org/corfudb/infrastructure/CorfuServer.java

```diff
@@ -196,17 +203,39 @@
                     + " --version                                                                "
                     + "              Show version\n";
 
-    private static volatile Thread corfuServerThread;
-    private static volatile Thread shutdownThread;
+    // Active Corfu Server.
+    private static volatile CorfuServer activeServer;
+
+    // Flag if set to true - causes the Corfu Server to shutdown.
+    private static volatile boolean shutdownServer = false;
+    // If set to true - triggers a reset of the server by wiping off all the data.
+    private static volatile boolean cleanupServer = false;
+    // Binds to all interfaces = inferred from the CLI parameters.
+    private static volatile boolean bindToAllInterfaces = true;
```

Comment:
Managing multiple sever instances within this Class is error-prone. Why can't the consuming code (i.e. main, jmh, etc...) just create multiple CorfuServer instances and manage its life-cycle?

Reply:
Fixed.